.github/workflows/python-module.yml

```diff
@@ -13,18 +13,19 @@ jobs:
         python-version: '3.6'
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        pip install setuptools wheel twine pytest
-        pip install -r requirements.txt
+        python -m pip install poetry
+        poetry install
+    - name: Check formatting
+      run: |
+        poetry run black --check wknml tests examples
     - name: Test
       run: |
         mkdir -p testoutput
-        pytest tests
+        poetry run pytest tests
     - name: Build and publish
       if: startsWith(github.event.ref, 'refs/tags')
       env:
-        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        PYPI_SCM_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        PYPI_SCM_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
       run: |
-        python setup.py sdist
-        twine upload dist/*
+        poetry publish --build -u $PYPI_SCM_USERNAME -p $PYPI_SCM_PASSWORD
```

Comment:
Did you check that this errors if one tries to overwrite a version which already exists? This would be important to not overwrite old versions by mistake.

Reply:
Yes, it should fail. See here (different project, but same command): https://github.com/hotzenklotz/pybeerxml/runs/1192257257?check_suite_focus=true#step:5:134